source/docs/software/sensors/limit-switch.rst

```diff
@@ -0,0 +1,53 @@
+Programming Limit Switches
+==========================
+
+Limit switches are often used to control mechanisms on robots. While limit switches are simple to use, they only can sense a single position of a moving part. This makes them ideal for ensuring that movement doesn't exceed some limit but not so good at controlling the speed of the movement as it approaches the limit. For example, a rotational shoulder joint on a robot arm would best be controlled using a potentiometer or an absolute encoder, the limit switch could make sure that if the potentiometer ever failed, the limit switch would stop the robot from going to far and causing damage.
+
+Values Provided
+---------------
+
+.. image:: images/using-limit-switches/image1.png
+
+Limit switches can have "normally opened" or "normally closed" outputs. Normally open means that the switch will report low (0 value) until the switch is closed and then it will output high (1 value).  Normally closed works in the opposite way (high until closed then low).  The usual way of wiring the switch is between a digital input signal connection and ground. The digital input has pull-up resistors that will make the input be high (1 value) when the switch is open, but when the switch closes the value goes to 0 since the input is now connected to ground. The switch shown here has both normally open and normally closed outputs.  To learn more about limit switch hardware see this :ref:`article <docs/hardware/sensors/proximity-switches:Mechanical Proximity Switches ("limit switches")>`.
+
+Controlling a Motor with Two Limit Switches
+-------------------------------------------
+
+.. tabs::
+
+   .. code-tab:: java
+
+     public void robotInit() {
+          var toplimitSwitch = new DigitalInput(0);
+          var bottomlimitSwitch = new DigitalInput(1);
+          PWMVictorSPX motor = new PWMVictorSPX(0);
+     }
+
+     public void teleopPeriodic() {
+          public void setMotorSpeed(double speed)) {
+               if (speed > 0) {
+                    if (toplimitSwitch.get()){
```

Comment:
These variables aren't accsssible here because they were declared in `robotInit()`'s scope.

Reply:
I am not very familiar with Timed.  What is the most common way that people get them in scope?

Reply:
I just removed the init stuff and people can infer them made however/wherever.

I could use some aid on the C++ version of this ;)

Reply:
@jasondaming I submitted a PR to your branch with C++ code and some cleanup on Java. 
I'm pretty sure that my C++ code is correct, but it should be reviewed by someone who actually knows C++ (@calcmogul?).

Reply:
Thanks a bunch! Merged.  @prateekma may be able to confirm also.